tests/flytekit/unit/extras/tensorflow/records/test_transformations.py

```diff
@@ -0,0 +1,116 @@
+from typing import Type
+
+import pytest
+import tensorflow as tf
+
+import flytekit
+from flytekit.configuration import Image, ImageConfig
+from flytekit.core import context_manager
+from flytekit.extras.tensorflow.record import (
+    TensorflowRecordFileTransformer,
+    TensorflowRecordsDirTransformer,
+    TFRecordDatasetConfig,
+)
+from flytekit.models.core.types import BlobType
+from flytekit.models.literals import Blob, BlobMetadata, Literal, Scalar
+from flytekit.models.types import LiteralType
+from flytekit.types.directory import TFRecordsDirectory
+from flytekit.types.file import TFRecordFile
+
+default_img = Image(name="default", fqn="test", tag="tag")
+serialization_settings = flytekit.configuration.SerializationSettings(
+    project="project",
+    domain="domain",
+    version="version",
+    env=None,
+    image_config=ImageConfig(default_image=default_img, images=[default_img]),
+)
+
+
+def write_to_tfrecord(path):
+    a = tf.train.Feature(bytes_list=tf.train.BytesList(value=[b"foo", b"bar"]))
+    b = tf.train.Feature(float_list=tf.train.FloatList(value=[1.0, 2.0]))
+    c = tf.train.Feature(int64_list=tf.train.Int64List(value=[3, 4]))
+    features = tf.train.Features(feature=dict(a=a, b=b, c=c))
+    example = tf.train.Example(features=features).SerializeToString()
+    with tf.io.TFRecordWriter(path) as writer:
+        for i in range(2):
+            writer.write(example)
+
+
+@pytest.mark.parametrize(
+    "transformer,python_type,format",
+    [
+        (TensorflowRecordFileTransformer(), TFRecordFile, TensorflowRecordFileTransformer.TENSORFLOW_FORMAT),
```

Comment:
Can you also add directory to this list? Also seems like you haven't used the right transformer name.